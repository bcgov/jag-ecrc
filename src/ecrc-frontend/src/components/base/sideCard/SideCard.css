.submitted-header {
  background-color: rgba(56, 89, 138, 1);
  border: 1px solid transparent;
  border-bottom: none;
  border-top-left-radius: 5px;
  border-top-right-radius: 5px;
  padding-left: 20px;
  padding-right: 20px;
  padding-bottom: 10px;
  color: #fff !important;
}

.move-heading {
  padding-left: 15px;
  font-weight: 600;
  font-style: normal;
  font-size: 16px;
  color: #ffffff;
  text-align: left;
  padding-top: 9px;
}

.heading-style {
  font-weight: 600;
  font-style: normal;
  font-size: 16px;
  color: #ffffff;
  text-align: left;
  padding-top: 9px;
}

.bluegrey-heading {
  padding-left: 15px;
  padding-bottom: 9px;
}

.bluegrey-heading h2 {
  margin-bottom: 0px;
}

.bluegrey-content {
  height: auto;
  background: inherit;
  background-color: rgba(241, 244, 250, 1);
  box-sizing: border-box;
  border-width: 4px;
  border-style: solid;
  border-color: rgba(255, 204, 0, 1);
  border-left: 0px;
  border-top: 0px;
  border-right: 0px;
  padding-top: 15px;
}

.content-style {
  padding-left: 15px;
  display: flex;
  flex-direction: column;
}

.container-background {
  background-color: rgba(56, 89, 138, 1);
  font-weight: 400;
  font-style: normal;
  font-size: 12px;
  text-align: left;
}

.submitted-header h2 {
  color: #fff !important;
}

.submitted-content {
  background-color: #eee;
  padding: 30px;
  margin-bottom: 20px;
}

.submit-container {
  background-color: #38598a;
  border-bottom: solid 4px #fc0;
  color: #fff;
  margin-bottom: 20px;
  padding: 15px;
}

.bluegrey-container {
  margin-bottom: 20px;
}

.submit-container h2 {
  color: #fff;
}

.dashboard-spacing {
  padding-left: 50px;
  padding-right: 50px;
  max-width: 400px;
}

.wide-dashboard-spacing {
  padding-left: 50px;
  padding-right: 50px;
  max-width: 520px;
}

img {
  max-width: 100%;
}

@media (max-width: 575.98px) {
  .dashboard-spacing {
    padding-left: 10px;
    padding-right: 10px;
  }
}

.contact-title {
  text-align: right;
  width: 80px;
  padding-right: 10px;
  font-weight: 600;
}
