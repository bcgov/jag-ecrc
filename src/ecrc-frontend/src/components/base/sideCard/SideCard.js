import React from "react";
import PropTypes from "prop-types";

import "./SideCard.css";

export default function SideCard({
  sideCard: { heading, content, type, image, imageLink }
}) {
  return (
    <div className="dashboard-spacing" style={{ position: "relative" }}>
      <div className="row">
        <div
          className="col-12"
          style={{ position: "relative", paddingTop: "30px" }}
        >
          <p />
          {type === "notice" && (
            <section
              id="notice-section"
              className="submit-container"
              style={{
                backgroundColor: "#F2F2F2",
                color: "#000",
                border: "none"
              }}
            >
              <h2 style={{ color: "#000" }}>{heading}</h2>
              <div className="submit-content">
                <p>{content}</p>
              </div>
            </section>
          )}
          {type === "blue" && (
            <section id="blue-section" className="submit-container">
              <h2 className="heading-style">{heading}</h2>
              <div className="submit-content">
                <p>{content}</p>
                {image && (
                  <a href={imageLink}>
                    <img
                      src={image}
                      alt="imagelink"
                      height="65px"
                      width="310px"
                    />
                  </a>
                )}
              </div>
            </section>
          )}
          {type === "bluegrey" && (
            <section id="bluegrey-section" className="bluegrey-container">
              <div className="container-background bluegrey-heading">
                <h2 className="heading-style">{heading}</h2>
              </div>
              <div className="bluegrey-content">
                <p className="content-style">{content}</p>
              </div>
            </section>
          )}
        </div>
      </div>
    </div>
  );
}

SideCard.propTypes = {
  sideCard: PropTypes.shape({
    heading: PropTypes.string.isRequired,
    content: PropTypes.array.isRequired,
    type: PropTypes.string.isRequired,
    image: PropTypes.string,
    imageLink: PropTypes.string
  }).isRequired
};
