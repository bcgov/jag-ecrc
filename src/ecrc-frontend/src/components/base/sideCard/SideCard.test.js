import React from "react";
import { create } from "react-test-renderer";
import { shallow } from "enzyme";
import SideCard from "./SideCard";

describe("SideCard Component", () => {
  const sideCardProps = {
    heading: "header",
    content: ["content"],
    type: "notice"
  };

  test("Renders the notice card when the type is notice", () => {
    const sideCard = shallow(<SideCard sideCard={sideCardProps} />);
    expect(sideCard.exists("#notice-section")).toEqual(true);
  });

  test("Renders the blue card when the type is blue", () => {
    const sideCard = shallow(
      <SideCard sideCard={{ ...sideCardProps, type: "blue" }} />
    );
    expect(sideCard.exists("#blue-section")).toEqual(true);
  });

  test("Renders the bluegrey card when the type is bluegrey", () => {
    const sideCard = shallow(
      <SideCard sideCard={{ ...sideCardProps, type: "bluegrey" }} />
    );
    expect(sideCard.exists("#bluegrey-section")).toEqual(true);
  });

  test("Renders as expected", () => {
    const sideCard = create(<SideCard sideCard={sideCardProps} />);
    expect(sideCard.toJSON()).toMatchSnapshot();
  });
});
