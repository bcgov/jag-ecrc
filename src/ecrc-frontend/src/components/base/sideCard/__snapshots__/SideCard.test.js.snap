// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideCard Component Renders as expected 1`] = `
<div
  className="dashboard-spacing"
  style={
    Object {
      "position": "relative",
    }
  }
>
  <div
    className="row"
  >
    <div
      className="col-12 pt-2"
      style={
        Object {
          "position": "relative",
        }
      }
    >
      <p />
      <section
        className="submit-container"
        id="notice-section"
        style={
          Object {
            "backgroundColor": "#F2F2F2",
            "border": "none",
            "color": "#000",
          }
        }
      >
        <h2
          style={
            Object {
              "color": "#000",
            }
          }
        >
          header
        </h2>
        <div
          className="submit-content"
        >
          <span>
            content
          </span>
        </div>
      </section>
    </div>
  </div>
</div>
`;
